src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -186,6 +217,28 @@ private async Task ProcessQueueAsync()
                     var context = await work.CreateRequestContextAsync(cancellationToken).ConfigureAwait(false);
                     if (work.MutatesServerState)
                     {
+                        if (CancelInProgressWorkUponMutatingRequest)
+                        {
+                            if (pendingTasks.Count > 0)
+                            {
+                                // Cancel all pending tasks
+                                var pendingTasksArray = pendingTasks.ToArray();
+                                for (var i = 0; i < pendingTasksArray.Length; i++)
```

Comment:
nit - could be a foreach

Reply:
This would complicate things a bit as the loop operates on the KeyValuePair Value while the Select statement below works against it's Key. I only want to get the enumeration from the ConcurrentDictionary once as doing so acquires all it's locks and if I did that twice, there would also be no guarantee that the two enumerations have the same items.